modules/vaos/app/controllers/vaos/v0/appointments_controller.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require_dependency 'vaos/application_controller'
+
+module VAOS
+  module V0
+    class AppointmentsController < ApplicationController
+      before_action :validate_params
+
+      def index
+        render json: each_serializer.new(appointments[:data], meta: appointments[:meta])
+      end
+
+      private
+
+      def appointment_service
+        VAOS::AppointmentService.for_user(current_user)
+      end
+
+      def appointments
+        @appointments ||=
+          appointment_service.get_appointments(type, start_date, end_date, pagination_params)
+      end
+
+      def each_serializer
+        "VAOS::#{params[:type].upcase}AppointmentsSerializer".constantize
+      end
+
+      def validate_params
+        raise Common::Exceptions::ParameterMissing, 'type' if params[:type].blank?
+        raise Common::Exceptions::InvalidFieldValue.new('type', params[:type]) unless %w[va cc].include?(params[:type])
```

Comment:
Nitpick: Should we make use of your `type` method here instead of using `params[:type]`?